src/components/CCPDashboard/PatientInfoTable.tsx

```diff
@@ -128,23 +139,79 @@ const EnhancedTableHead = (props: EnhancedTableProps) => {
 export const PatientInfoTable = ({
   patients,
   type,
+  eventId,
+  ccpId,
 }: {
   patients: Patient[];
   type: CCPDashboardTabOptions;
+  eventId: string;
+  ccpId: string;
 }) => {
   const classes = useStyles();
   const [order, setOrder] = React.useState<Order>('desc');
   const [orderBy, setOrderBy] = React.useState<string>(
     type === CCPDashboardTabOptions.Hospital ? 'transportTime' : 'updatedAt'
   );
   const [openDetails, setOpenDetails] = React.useState(false);
+  const [openDeletePatient, setOpenDeletePatient] = React.useState(false);
   const [selectedPatient, setSelectedPatient] = React.useState(null);
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const history = useHistory();
+
+  const [deletePatient] = useMutation(DELETE_PATIENT, {
+    update(cache, { data: { deletePatient } }) {
+      if (!deletePatient) {
+        return;
+      }
+      let { patients } = cache.readQuery<any>({
+        query: GET_ALL_PATIENTS,
+      });
+      const filtered = patients.filter(
+        (patient) => patient.id !== ((selectedPatient as unknown) as Patient).id
+      );
+      patients = filtered;
+      cache.writeQuery({
+        query: GET_ALL_PATIENTS,
+        data: { patients },
+      });
+    },
+  });
 
   const handleOpenDetails = (patient) => {
     setSelectedPatient(patient);
     setOpenDetails(true);
   };
 
+  const handleClickOptions = (event, patient) => {
+    setSelectedPatient(patient);
+    event.stopPropagation();
+    setAnchorEl(anchorEl ? null : event.currentTarget);
+  };
+
+  const handleCloseMenu = () => {
+    setAnchorEl(null);
+  };
+
+  const handleClickEdit = () => {
+    history.replace(
```

Comment:
I think this should be `history.push`.

For reference:
push(path, [state]): (function), pushes a new entry onto the history stack
replace(path, [state]): (function), replaces the current entry on the history stack

`replace` would prevent you from accessing the previous url when you click the browser "back" button.